test/e2e/autotls/auto_tls_test.go

```diff
@@ -21,32 +21,55 @@ import (
 	"context"
 	"crypto/tls"
 	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
 	"net/http"
 	"testing"
 	"time"
 
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/test/ingress"
 	"knative.dev/serving/pkg/apis/networking"
 	routenames "knative.dev/serving/pkg/reconciler/route/resources/names"
 	"knative.dev/serving/test"
 	testingress "knative.dev/serving/test/conformance/ingress"
 	"knative.dev/serving/test/e2e"
 	v1test "knative.dev/serving/test/v1"
+
+	"github.com/kelseyhightower/envconfig"
+
+	"google.golang.org/api/dns/v1"
 )
 
 const (
-	systemNamespace = "knative-serving"
+	dnsRecordDeadline = 600 // 10 min
 )
 
+type dnsRecord struct {
+	ip     string
+	domain string
+}
+
+type config struct {
+	DnsZone                       string `split_words:"true" required:"false"`
```

Comment:
```suggestion
	DNSZone                       string `split_words:"true" required:"false"`
```

Reply:
I have to use `DnsZone` here so that it can be mapped to environment variable `DNS_ZONE`. This is a usage of "github.com/kelseyhightower/envconfig"